# Ejercicio completo de Data Frames

Carga en Python el CSV llamado run.csv que encontrarás en el repositorio del curso.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../../data/run.csv")
df["variacion_pulso"] = (df["pulso.despues"] - df["pulso.antes"])/df["pulso.antes"]

In [3]:
df.head()

,Unnamed: 0,pulso.antes,pulso.despues,hace.deporte,fuma,genero,altura,peso,tipo.actividad,variacion_pulso
0,1,48,54,no,si,H,68.0,150,nula,0.125000
1,2,54,56,no,si,H,69.0,145,moderada,0.037037
2,3,54,50,no,no,H,69.0,160,moderada,-0.074074
3,4,58,70,si,no,H,72.0,145,moderada,0.206897
4,5,58,58,no,no,H,66.0,135,intensa,0.000000


## Pregunta 1

Indica cuántos estudiantes formaron parte del estudio de deporte.

In [4]:
df.shape[0]

92

El estudio lo formaron 92 estudiantes.

## Pregunta 2

Indica cuántos individuos son hombres y cuántos son mujeres.

In [5]:
len(df[df["genero"]=="H"])

57

Hay 57 hombres.

In [6]:
len(df[df["genero"]=="M"])

35

Hay 35 mujeres.

## Pregunta 3

Calcula el porcentaje medio de variación del pulso por minuto entre antes y después de hacer ejercicio y compara el valor de los que hacen ejercicio habitualmente y los que no. ¿Observas mucha diferencia?

In [7]:
df.groupby("hace.deporte")["variacion_pulso"].mean()

hace.deporte
no    0.002095
si    0.262154
Name: variacion_pulso, dtype: float64

Existe mucha diferencia. Las personas que hacen deporte tienen una media de 26.22 % de variación en las pulsaciones por minuto. Mientras que las personas que no hacen deporte habitualmente es de tan solo un 0.21 %.

## Pregunta 4

Calcula el porcentaje medio de variación del pulso por minuto entre antes y después de hacer ejercicio para los estudiantes que hacen ejercicio habitualmente y compara el valor de los hombres con el de las mujeres. ¿Observas mucha diferencia?

In [8]:
df[df["hace.deporte"]=="si"].groupby("genero")["variacion_pulso"].mean()

genero
H    0.191769
M    0.415722
Name: variacion_pulso, dtype: float64

La diferencia más significativa es en las personas que sí realizan deporte habitualmente. Los hombres tienen una variación media del 41.57 %, mientras que la de las mujeres es del 19.18 %, más de la mitad que los hombres.

## Pregunta 5

Calcula el porcentaje medio de variación del pulso por minuto entre antes y después de hacer ejercicio para los estudiantes que no hacen ejercicio habitualmente y compara el valor de los fumadores con los no fumadores. ¿Observas mucha diferencia?

In [9]:
df[df["hace.deporte"]=="no"].groupby(["hace.deporte", "fuma"])["variacion_pulso"].mean()

hace.deporte  fuma
no            no     -0.000648
              si      0.009122
Name: variacion_pulso, dtype: float64

La variación media de las pulsaciones por minuto de las personas que no hacen ejercicio habitualmente es baja, aunque en fumadores (0.91 %) es significativamente superior a los no fumadores (-0.0065 %).

## Pregunta 6

Calcula el porcentaje medio de variación del pulso por minuto entre antes y después de hacer ejercicio de todos los estudiantes según el tipo de actividad física que realizan. ¿Observas alguna diferencia?

In [10]:
df.groupby("tipo.actividad")["variacion_pulso"].mean()

tipo.actividad
intensa     0.013475
moderada    0.140157
nula        0.125000
suave       0.037472
Name: variacion_pulso, dtype: float64

La actividad moderada es la que más variación de pulso representa con un 14.02 %. En tipos de actividades nulas el porcentaje de media de pulsaciones es ligeramente inferior con un 12.5 %. En actividades suaves e intensas las variaciones de pulso no son representativas.